qiskit_experiments/data_processing/nodes.py

```diff
@@ -192,45 +216,42 @@ def _process(
         Raises:
             DataProcessorError: If the SVD has not been previously trained on data.
         """
-
         if not self.is_trained:
             raise DataProcessorError("SVD must be trained on data before it can be used.")
 
-        n_qubits = datum.shape[0] if len(datum.shape) == 2 else datum.shape[1]
-        processed_data = []
-
-        if error is not None:
-            processed_error = []
+        # IQ axis is reduced by projection
+        if self._n_shots == 0:
+            # level1 average mode
+            dims = self._n_circs, self._n_slots
         else:
-            processed_error = None
+            # level1 single mode
+            dims = self._n_circs, self._n_shots, self._n_slots
 
-        # process each averaged IQ point with its own axis.
-        for idx in range(n_qubits):
+        processed_data = np.zeros(dims, dtype=float)
+        error_vals = np.zeros(dims, dtype=float)
 
+        for idx in range(self._n_slots):
+            scale = self.scales[idx]
             centered = np.array(
                 [datum[..., idx, iq] - self.means(qubit=idx, iq_index=iq) for iq in [0, 1]]
             )
+            angle = np.arctan(self._main_axes[idx][1] / self._main_axes[idx][0])
 
-            processed_data.append((self._main_axes[idx] @ centered) / self.scales[idx])
+            processed_data[..., idx] = (self._main_axes[idx] @ centered) / scale
 
             if error is not None:
-                angle = np.arctan(self._main_axes[idx][1] / self._main_axes[idx][0])
-                error_value = np.sqrt(
-                    (error[..., idx, 0] * np.cos(angle)) ** 2
-                    + (error[..., idx, 1] * np.sin(angle)) ** 2
+                error_vals[..., idx] = (
+                    np.sqrt(
+                        (error[..., idx, 0] * np.cos(angle)) ** 2
+                        + (error[..., idx, 1] * np.sin(angle)) ** 2
+                    )
+                    / scale
                 )
-                processed_error.append(error_value / self.scales[idx])
 
-        if len(processed_data) == 1:
-            if error is None:
-                return processed_data[0], None
-            else:
-                return processed_data[0], processed_error[0]
+        if self._n_circs == 1:
+            return processed_data[0], error_vals[0]
 
-        if error is None:
-            return np.array(processed_data), None
-        else:
-            return np.array(processed_data), np.array(processed_error)
+        return processed_data, error_vals
```

Comment:
What happens when `error is None`? Looks like this will return 0 (or np.zeros) error which is not supposed to happen.

Reply:
Reverted to original code https://github.com/Qiskit/qiskit-experiments/pull/476/commits/a7e7b9c721e2ab3f9fdf183852e4c9186de152ed